i3ipc.py

```diff
@@ -413,6 +413,15 @@ def __init__(self, data, parent, conn):
             else:
                 setattr(self, attr, None)
 
+        # Make sure marks is never None (use []) to simplify client code. Older
+        # versions of i3 used mark instead of marks, so collapse that into
+        # marks to provide a consistent API.
+        if not self.marks:
```

Comment:
Yes, that would be good if you removed it. I like [git undo](https://github.com/tj/git-extras/blob/master/Commands.md#git-undo) from git-extras for that.

Also, you should be doing pull requests in a topic branch not the master branch.